src/datastore/parser.hpp

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <fstream>
+#include <fstream>
+#include <stdexcept>
+#include <vector>
+#include <iostream>
+#include <sstream>
```

Comment:
Let's move all the headers that aren't required to use these functions into the `.cpp` file.